### PR TITLE
[SPARK-49173][CONNECT] Change Spark Connect shell prompt from `@` to `scala>`

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to change the shell prompt from `@` to `scala>` in Spark Connect shel.

### Why are the changes needed?

To be consistent with Spark Classic.

### Does this PR introduce _any_ user-facing change?

Yes. Before:

```
@ ...
```

now

```
scala> ...
```

### How was this patch tested?

Manually tested on the top of https://github.com/apache/spark/pull/47402

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
This PR is dependent on https://github.com/apache/spark/pull/47402

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/application/ConnectRepl.scala [hunk 1]
```diff
@@ -137,14 +141,64 @@ Spark session available as 'spark'.
...
+              s"""
+        val ${b.name} = ammonite
+          .repl
+          .ReplBridge
+          .value
+          .Internal
+          .replArgs($idx)
+          .value
+          .asInstanceOf[${b.typeName.value}]
+        """
```
Comment (pan3793) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/application/ConnectRepl.scala:171]: the indent looks weird

Comment (HyukjinKwon) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/application/ConnectRepl.scala:171]: It's done by scalafmt :-).

**Pull request closed without merging**